New template for FTB Modpacks based on Fabric

Linx-ESP opened the pull request:
It might be possible to combine it with the FTB-Forge template, but it isn't directly possible since Forge uses a couple extra steps and java arguments.

~~If the selector for Garbage Collector is unwanted, I will delete it. Right now is a toggle between G1GC an ShenandoahGC.~~
Changed and tweaks as per [this](https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks#shenandoah)

Linx-ESP committed:
Create minecraft-ftb-faabric,json

Linx-ESP committed:
Rename minecraft-ftb-faabric,json to minecraft-ftb-fabric,json

Linx-ESP committed:
Create README.md

Linx-ESP committed:
Update minecraft-ftb-fabric,json

Linx-ESP committed:
Update minecraft-ftb-fabric,json

Linx-ESP committed:
Rename minecraft-ftb-fabric,json to minecraft-ftb-fabric.json

LordRalex commented on minecraft-ftb-fabric/minecraft-ftb-fabric.json:
```diff
@@ -0,0 +1,133 @@
+    {
+      "commands": [
+        "chmod u+x linux",
+        "./linux ${modpack_id} ${modpack_version} --auto --noscript"
```
iirc, the wrapper file they download above actually includes those flags, so we don't need an explicit CLI arg here for it.

Linx-ESP replied:
You mean `./linux ${modpack_id} ${modpack_version} --auto --noscript` should not have `--auto --noscript`?  
In the other two FTB templates it is the same way, so it would require changing them there too.

LordRalex replied:
The `${modpack_id} ${modpack_version}` is not required in the call, because the binary already contains the ids.

Linx-ESP replied:
Changed to `"./linux --auto --noscript"`
Would you like me to change the other FTB templates (on another PR)?

Linx-ESP replied:
It does need to be `./linux ${modpack_id} ${modpack_version} --auto --noscript` or it will error with 

> Starting process: ./linux --auto --noscript
> Server installer version 22.1213.2347-linux-amd64 commit fba9792cf71889ce973a6a0846a194e3113e1678
>                       _                  _              _     
>                      | |                | |            | |    
> 
>  based on commit fba9792cf71889ce973a6a0846a194e3113e1678
> 
> Usage:
>   linux <modpackid> <versionid> - will install the modpack specified by <modpackid> with the version specified by <versionid>
>   linux <modpackid> - will install the modpack specified by <modpackid> with the latest version available
> 
>   _ __ ___   ___   __| |_ __   __ _  ___| | _____   ___| |__  
>  | '_ ` _ \ / _ \ / _` | '_ \ / _` |/ __| |/ / __| / __| '_ \ 
>  | | | | | | (_) | (_| | |_) | (_| | (__|   <\__ \| (__| | | |
>  |_| |_| |_|\___/ \__,_| .__/ \__,_|\___|_|\_\___(_)___|_| |_|
>                        | |                                    
>                        |_|                                    
>  modpacks.ch server downloader golang - build 22.1213.2347-linux-amd64Arguments:
> 2023/03/06 19:42:00 Cannot locate modpack via filename. Error: unable to parse filename: linux
>   --auto - Ask no questions, use defaults.
>   --path - Directory to install in. Default: current directory
>   --noscript - Skip creating start script. Default: false
>   --nojava - Skip downloading a compatible Adoptium JRE. Default: false
>   --threads - Number of threads to use for downloading. Default: cpucores * 2Do a full integrity check, even on files not changed by the update. integrityupdate assumed. Default: true
> 
>   --integrityupdate - Whether changed files should be overwritten with fresh copies when updating. Most useful when used with Auto. Default: false
>     Example: You changed config/test.cfg on your server from default. The modpack updates config/test.cfg - with this flag, it will assume you wish to overwrite with the latest version
>   --integrity -   --latest -   --verbose - Be a bit noisier on actions taken. Default: false
> Install latest, ignoring any version in the file name or arguments. Default: false
>   --curseforge - Specifies that pack is a Curseforge modpack
>   --help - This help"
>

LordRalex commented on minecraft-ftb-fabric/minecraft-ftb-fabric.json:
```diff
@@ -0,0 +1,133 @@
+      "required": true,
+      "value": "0.0.0.0"
+    },
+    "javaversion": {
```
Watch the version here, because 8 doesn't work for new versions. I'd probably say assume newer packs, so a newer version of Java so it doesn't "break" immediately.

(I wish APIs had the versions better....)

Linx-ESP replied:
Changed default to 17

Linx-ESP committed:
Update minecraft-ftb-fabric.json

Linx-ESP committed:
Update minecraft-ftb-fabric.json

Linx-ESP committed:
update java args

Removed Shenandoah as per https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks#shenandoah.  
Tweaked G1GC as per https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks#shenandoah.  
Changed "users can edit" from modpack ids.

Linx-ESP commented on minecraft-ftb-fabric/minecraft-ftb-fabric.json:
```diff
@@ -0,0 +1,119 @@
+  ],
+  "run": {
+    "stop": "stop",
+    "command": "java${javaversion} -Xmx${memory}M -Xms${memory}M -XX:+UseG1GC -XX:MaxGCPauseMillis=130 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=28 -XX:G1HeapRegionSize=16M -XX:G1ReservePercent=20 -XX:G1MixedGCCountTarget=3 -XX:InitiatingHeapOccupancyPercent=10 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=0 -XX:Surv ...
```
Removed Shenandoah as per https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks#shenandoah.  
Tweaked G1GC as per https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks#server-g1gc.
